src/core/Akka.TestKit/TestKitBase_AwaitAssert.cs

```diff
@@ -53,6 +53,7 @@ public async Task AwaitAssertAsync(Action assertion, TimeSpan? duration=null, Ti
                 cancellationToken.ThrowIfCancellationRequested();
                 try
                 {
+                    // TODO: assertion can run forever, need a way to stop this if this happens.
```

Comment:
Probably isn't a good way to do this without cancellation support or some crazy detached task execution structure. Better to let the test runner just hang and make it the users' problem to debug.